input.c

```diff
@@ -2562,6 +2566,55 @@ input_osc_4(struct input_ctx *ictx, const char *p)
 	free(copy);
 }
 
+/* Handle the OSC 8 sequence for writing hyperlinks. */
+static void
+input_osc_8(struct input_ctx *ictx, const char *p)
+{
+  struct grid_cell *gc = &ictx->cell.cell;
+  char *id_;
+  char *end;
+  struct hyperlink *hl;
+  int id;
+  u_int i;
+
+  if((end = strchr(p,';')) == NULL) {
+		log_debug("%s: bad OSC 8: %s", __func__,p);
+    return;
+  }
+  /* parse ID and check pane tree, reuse entry if present */
+  id_=strstr(p,"id=");
+  id=0;
+  if(id_!=NULL&&id_<end){
+    id_+=3;
+    while(*id_>='0' && *id_<='9'){
+      id = id * 10 + *id_++ - '0';
+    }
+  }
+
+  end++;
+  if(*end == '\0'){
+    gc->link = 0;
+    return;
+  }
+
+  log_debug("%s: OSC 8 id=%d", __func__, id);
+  hl = server_get_hyperlink(id);
+  if(hl==NULL) {
+    hl = xcalloc(1, sizeof *hl);
+    if(id!=0){
+      hl->id = id;
+    }else{
+      hl->id = ++next_hyperlink;
+    }
+    hl->link = xstrdup(end);
```

Comment:
Is it a good idea to update the link of existing entry to the latest one with the same id?